### DISCUSSION TITLE
Push Notification

### DISCUSSION THREAD

**Discussion by TrustworthyDev** [community/community#140178] [Programming Help] [answered]
### Select Topic Area

Question

### Body

I am trying to build the Push Notification into my a Mobile app, which is built with Vue.js for frontend and Laravel for backend.
And also, I am planning to use the online service in Firebase.
If you have any experience of it, share your experience or idea please.

**Comment by tshandler** [comment]
Hi,
I have some of experience in that you described.
Need more conversation about it.

**Reply by TrustworthyDev** [reply]
What and how do I have to do for this?

**Comment by tshandler** [comment]
What framework or tool did you use to build your mobile app?

**Reply by TrustworthyDev** [reply]
Ionic (Vue.js)!

**Comment by tshandler** [comment]
All development environment is created and don't need anymore?

**Reply by TrustworthyDev** [reply]
Yeah, all conditions are prepared, which includes the developing and building.
I can build apps for both Android or iPhone.
Only what I need is to implement the Push Notification into my Ionic Mobile app.

**Comment by tshandler** [comment, marked as answer]
Okay, everything is clear.
To implement push notifications in your Ionic (Vue.js) mobile app using Firebase, follow the steps:
1. Set up Firebase project.
Main Point is to register your app into Firebase with bundle ID.

2. Install Firebase and Firebase Messaging
Main Point is to install essential node modules into your Ionic app.

3. Configure Firebase in Ionic
Add the configuration into app.vue of your Ionic project.

4. Handle Push Notification with Capacitor
Need to handle notification using Capacitor's Push Notification API.

5. Setup Firebase Cloud Messaging (FCM)
Enable Cloud Messaging in your project settings.

Finally, Make sure to test your app on physical devices (iOS and Android) because push notifications may not work in the simulator or emulator.

I hope these steps will be helpful to you.
